### PR TITLE
Tennis players basic info.

### PR DESCRIPTION
**Pull request by thecoder12**
Tennis players basic info.

### CODE REVIEW

**Comment by javathunderman** [comment]
@thecoder12 Hey! Thanks for contributing. 

Just letting you know, the tennis player basic info instant answer that's in your pull request is already handled by the Wikipedia Instant Answer, and it is not necessary to have a separate instant answer for this purpose specifically. 

I left some comments on your quick start goodie for you to fix, and it should be ready to merge. 

cc @mintsoft

**Comment by javathunderman** [comment]
We'd also prefer if you could use the pull request template (https://github.com/duckduckgo/zeroclickinfo-goodies/blob/master/pull_request_template_goodie.md)
and fill it out in a comment; it helps us see what your instant answer is about.

File: lib/DDG/Goodie/IsAwesome/TennisCoder.pm [hunk 1]
```diff
@@ -0,0 +1,33 @@
+package DDG::Goodie::IsAwesome::TennisCoder;
+# ABSTRACT: Write an abstract here
+# Start at https://duck.co/duckduckhack/goodie_overview if you are new
+# to instant answer development
+
+use DDG::Goodie;
+
```
Comment (javathunderman) [inline lib/DDG/Goodie/IsAwesome/TennisCoder.pm]: Could you put "use strict;" here? It helps Github recognize the language as Perl5 and not Perl6.

File: lib/DDG/Goodie/IsAwesome/TennisCoder.pm [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+zci answer_type => "is_awesome_tennis_coder";
+zci is_cached   => 1;
+
+# Metadata.  See https://duck.co/duckduckhack/metadata for help in filling out this section.
```
Comment (javathunderman) [inline lib/DDG/Goodie/IsAwesome/TennisCoder.pm]: Please remove these comments.

File: lib/DDG/Goodie/IsAwesome/TennisCoder.pm [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+name "IsAwesome TennisCoder";
+description "Succinct explanation of what this instant answer does";
```
Comment (javathunderman) [inline lib/DDG/Goodie/IsAwesome/TennisCoder.pm]: Please fill out these metadata fields- it helps us organize instant answers better.

File: lib/DDG/Goodie/IsAwesome/TennisCoder.pm [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+primary_example_queries "first example query", "second example query";
+secondary_example_queries "optional -- demonstrate any additional triggers";
+# Uncomment and complete: https://duck.co/duckduckhack/metadata#category
+# category "";
+# Uncomment and complete: https://duck.co/duckduckhack/metadata#topics
+# topics "";
+code_url "https://github.com/duckduckgo/zeroclickinfo-goodies/blob/master/lib/DDG/Goodie/IsAwesome/TennisCoder.pm";
+attribution github => ["GitHubAccount", "Friendly Name"],
```
Comment (javathunderman) [inline lib/DDG/Goodie/IsAwesome/TennisCoder.pm]: Please fill in these attribution links.

**Comment by thecoder12** [comment]
You are right, "About" section does cover up the basic info. I'll take care of the other pre-conditions for pushing code to prod.
Thanks!!!

**Comment by javathunderman** [comment]
No problem, let us know if you have problems with the code.

**Comment by thecoder12** [comment]
Could you please point to the resolution of below issue,
https://gist.github.com/MrChrisW/794e24904126c0c5199b

**Comment by javathunderman** [comment]
Ah, I let @moollaza know about that. We made some changes to the DuckPAN application that we use to test instant answers and it should be fixed soon. 

If you could commit the code, I'll take another look at it and see if it's OK to be merged.

**Comment by javathunderman** [comment]
@thecoder12 There's a temporary fix you could use now: `duckpan DDG::GoodieBundle::OpenSourceDuckDuckGo`, and then you can run the `duckpan upgrade` , `duckpan ddg` and `duckpan server` commands. Let me know if you need any more help. Again, you can commit the code to Github anyway and I can take a look at it.

**Comment by thecoder12** [comment]
Thanks...I will check the fix and let you know the status.

**Comment by thecoder12** [comment]
It didnt work....Please let me know if I'm going in wrong direction. 

[03:42 PM codio@bonjour-pump zeroclickinfo-goodies {master}]$ duckpan DDG::GoodieBundle::OpenSourceDuckDuckGo                                                 
You already have latest version (1069) of DDG::GoodieBundle::OpenSourceDuckDuckGo                                                                             
[03:42 PM codio@bonjour-pump zeroclickinfo-goodies {master}]$ duckpan upgrade                                                                                 
Emptying DuckPAN cache...                                                                                                                                     
DuckPAN cache emptied                                                                                                                                         
You already have latest version (0.181) of App::DuckPAN                                                                                                       
You have an older version (0.134) of DDG than duckpan.org. Installing latest version (0.136)                                                                  
--> Working on http://duckpan.org/authors/id/M/MO/MOOLLAZA/DDG-0.136.tar.gz                                                                                   
Fetching http://duckpan.org/authors/id/M/MO/MOOLLAZA/DDG-0.136.tar.gz ... OK                                                                                  
Configuring DDG-0.136 ... OK                                                                                                                                  
Building and testing DDG-0.136 ... FAIL                                                                                                                       
! Installing http://duckpan.org/authors/id/M/MO/MOOLLAZA/DDG-0.136.tar.gz failed. See /home/codio/.cpanm/work/1431963738.5014/build.log for details. Retry wit
h --force to force install it.                                                                                                                                
[03:42 PM codio@bonjour-pump zeroclickinfo-goodies {master}]$

**Comment by thecoder12** [comment]
I'm thinking of changing the Tennis ZCI...once i finalize it...i will surely push the code and let you know accordingly.
Thanks!!!

**Comment by javathunderman** [comment]
@thecoder12 Looks like that fix doesn't work anymore :-1: 

But if you push the new code and ping me, I can take another look at the code.

**Comment by thecoder12** [comment]
Sure. Not a problem.

**Comment by javathunderman** [comment]
Also, if you could fill out the pull request template here: (https://github.com/duckduckgo/zeroclickinfo-goodies/blob/master/pull_request_template_goodie.md) and just paste it into another comment, that would be great.

**Comment by MrChrisW** [comment]
@javathunderman Thanks for your help with the review. 

@thecoder12 I think these type of queries are already handled by the [Wikipedia IA](https://duck.co/ia/view/wikipedia_fathead), so unfortunately I'm not sure we can use this Goodie!

Examples:
[Novak Djokovic](https://duckduckgo.com/?q=roger+federer&t=ffcm&ia=about)

[Roger Federer](https://duckduckgo.com/?q=novak+djokovic&t=ffcm&ia=about)

//CC @moollaza

**Comment by thecoder12** [comment]
Sure. I realized that much later, that its been handled in About section. I
am thinking of revamping my code as per the Tennis Stats which doesn't
exists.
BTW, regarding the About section for especially "Roger Federer" ,seems a
bit misleading. As the player plays both doubles and Singles, there seems
to be no demarcation between the two categories. So his ranking is shown 2
& 24 without knowing the category. I am just pointing this out so that i
could know the way where we can directly pull the wiki table[specially the
right-most] section, so that we won't need to use reg-ex to process the
HTML code as per ZCI-About.
Please forgive my understanding if I am missing out something.

**Comment by MrChrisW** [comment]
@thecoder12 

>Sure. I realized that much later, that its been handled in About section. I am thinking of revamping my code as per the Tennis Stats which doesn't exists.

That's absolutely fine, you might want to look at using a [Spice](https://duck.co/duckduckhack/determine_your_instant_answer_type) to handle the Tennis Stats as this would allow updated information to be displayed - as you'd use an API to pull the stats.

However you'll need to find an appropriate API - https://duckduckgo.com/?q=tennis+stats+api

______

>BTW, regarding the About section for especially "Roger Federer" ,seems a bit misleading. As the player plays both doubles and Singles, there seems to be no demarcation between the two categories. So his ranking is shown 2 & 24 without knowing the category. I am just pointing this out so that i could know the way where we can directly pull the wiki table[specially the right-most] section, so that we won't need to use reg-ex to process the HTML code as per ZCI-About.

That looks like a :bug: bug to me! That will need to be reported to DDG either via https://duck.co/feedback/bug/-

No distinction between double and single ranking:
selection_420

>Please forgive my understanding if I am missing out something.

No problem! Always feel free to ask questions, we're here to help!

**Comment by moollaza** [comment]
Hey guys, sorry I'm a little late to the conversation. @MrChrisW is correct that we support this kinds of queries for the most part via Wikipedia.

@thecoder12 I really wish you would have written in to us ahead of time, because then I would have been able to tell you that we're actually already underway developing Sports IA's internally for most major leagues. This means eventually we'll have player bios, very similar to what you've done here and it will be coming from an API which means the data will always be up to date.

Using static text like this isn't ideal because then it requires maintenance and would require manual data entry.

So, unfortunately I'm going to have to close this PR. I'm **really** sorry to do this, but I hope you can understand why.

Is there anything else you'd like to work on? We have plenty of bugs that need fixing, improvements that need implementing and of course an ideas forum full of potential Instant Answers. If you're making something new though, please email (or msg via irc/gittr) us first so we can make sure you're heading in the right direction.

**Comment by javathunderman** [comment]
@moollaza this also has a Quickstart goodie, FYI.

**Comment by moollaza** [comment]
@javathunderman ah sorry I missed that. @thecoder12 if you want we can re-open this and you can remove all the other files not related to your `IsAwesome`  Goodie, or you can make a new pull request.

**Pull request closed without merging**